### ISSUE TITLE
GSConnect could not pair two computers, is it a bug or a limitation?

### ISSUE THREAD

**Issue by deltasata** [GSConnect/gnome-shell-extension-gsconnect#1884]
### Describe the bug

I have two laptops, both on Ubuntu 24.04, Gnome 46 and have GSConnect. I could see one from the other, but pressing 'pair' does nothing, does not send any request to the other! Do I have to do something? Or, does GSConnect work between two computers?

### GSConnect version

58

### Installed from

Extension Manager

### GNOME Shell version

46

### Linux distribution/release

Ubuntu 24.04.1

### Paired device(s)

Another Ubuntu 24.04 laptop

**Comment by uncharted-user** [comment]
I also experienced that issue. Is that a limitation? I tried to connect between Fedora 41 with Windows 10.

**Comment by ferdnyc** [comment]
@uncharted-user 

Windows support, in particular, is _not_ very well tested... but I was able to pair a Windows 10 system running KDE Connect with GSConnect on my Fedora desktop, relatively recently.

I do remember that my Windows machine didn't see the Fedora box _at all_, initially, and since there doesn't appear to be any way to initiate a manual connection from the Windows side, I ended up using "Add device by IP..." in GSConnect to link it with the Windows box. That made the Linux box show up in Windows, and I proceeded with the pairing from there.

I can't promise anything _else_ will work between GSConnect and a Windows machine, but pairing should at least be possible.

If you're having trouble getting messages (including pairing requests) through from one system to the other, it's possibly a firewall issue on at least one end. Both systems need to have ports opened in their respective firewalls, in order to communicate. (And for discovery to work, they need to be connected via a network that will let them hear each other's broadcast traffic.)

(When I installed KDE Connect on Windows and launched it for the first time, I remember Windows 10 threw up a firewall access dialog where I was able to give it permission to communicate over private and/or public networks.)

**Comment by ferdnyc** [comment]
@uncharted-user

Hmm. If KDE Connect on Windows was working with Android devices, then the firewall setup sounds like it was correct at least on that end. (Actually, if both devices could see the other, unpaired, that's a good sign for firewall setup all around.)

Since you have Android devices in the mix, were you able to pair any of _those_ with GSConnect on your Ubuntu system?

If **not**, then the problem is simply GSConnect not being able to pair with devices. In which case, the [Generate Support Log](https://github.com/GSConnect/gnome-shell-extension-gsconnect/wiki/Help#generate-support-log)<sup>1</sup> output collected when trying to pair the devices might be helpful.

If **yes** (you _can_ pair Android devices, but not Windows), then I'm stumped. But, Generate Support Log could still be helpful.

#### Notes

1. As the wiki entry says, the process is:
    1. activate "Generate Support Log"
    2. _THEN_ **try to reproduce the problem you're seeing** (IOW, try to pair the devices, from both ends — in fact, even better if you wait to _connect_ the other device to the network until after you've activated Generate Support Log, so we can see the log of device discovery as well.)
    3. Once you've reproduced the (non-)pairing, click "Review Log" to get the collected debug logs for the session.

**Comment by uncharted-user** [comment]
>Since you have Android devices in the mix, were you able to pair any of those with GSConnect on your Ubuntu system?
Yes, the only problem was there was no way to send files from Ubuntu to Windows 10.

>But, Generate Support Log could still be helpful.
I will once I can manage some free time.
